### PR TITLE
Add bridge extension

### PR DESCRIPTION
**Pull request by shaftoe**
**Branch:** `newBridge` → `develop`
**Handoff:** see the /pi 🤖 Handoff comment on this PR.

### CODE REVIEW

**Comment by shaftoe** [comment]
/pi 🤖 Handoff from local session

## TITLE
Add §9 platform/forgejo_url config override to Bridge.create

## Done
- Added `apiBaseUrlFor(serverUrl, platformType)` pure helper to `bridge.ts` — forces `/api/v1` for forgejo/codeberg regardless of URL substring (fixes the generic-hostname gap), defers to shared `apiBaseUrlFromServerUrl` for github
- Refactored `createOctokit` to accept `platformType` (3-arg, routes through `apiBaseUrlFor`) — only internal callers, all updated
- Wired `config?: Pick<BridgeConfig, 'platform' | 'forgejo_url'>` into `Bridge.create` — explicit platform forces platformType + known-host status (rescues generic-hostname forges); `forgejo_url` overrides remote-derived serverUrl only under an explicit platform; stored `discovery` reflects effective post-override values
- Moved config loading before `Bridge.create` in `index.ts` and passes it in; `auto_sync` still feeds enrichment hook
- Updated `config.ts` and `CONSTITUTION.md` docstrings to drop stale "not consumed yet / later phase" language; removed completed TODO bullet
- Added 13 tests: `apiBaseUrlFor` override + no-regression cases, 3-arg `createOctokit`, 6 `Bridge.create` override cases (forgejo/codeberg/github rescue, forgejo_url override, auto-doesn't-rescue guard, override-still-requires-token guard)
- Committed as `798bfd9` on `newBridge`; `bun run validate` green, 151 tests pass

## Next
The remaining CONSTITUTION TODO is edge-case testing: **fork PRs** (the `pulls.list` `head` filter in `resolveCurrentPR` is same-owner only, so cross-owner fork PRs are silently missed) and **expired tokens mid-session** (401/403 from `octokit.pulls.create` is caught in `/handoff`, but the read-only tools' error UX on a stale token is unverified). After that, push the branch to get a clean CI signal — the latest commit is non-`[skip ci]` so CI will run validate + tests against the full tree. The branch has not been pushed yet (local only).

**Comment by shaftoe** [comment]
/pi 🤖 Handoff

## Done

- Addressed remaining CONSTITUTION TODO — edge-case testing for **fork PRs** and **expired tokens mid-session**
- **Fork PRs:** Added tests documenting the   filter limitation (cross-owner fork PRs silently missed) — tested via real method in  and 
- **Expired tokens:** Added  with  helper; updated  and  to catch 401/403 and return actionable error messages
- Added 6 new tool tests; rebased to include additional handoff tests (162 → 168 total passing)
- Updated CONSTITUTION.md TODO section (marked edge-case testing as complete)
- Pushed commit `10e99b5` (non-`[skip ci]`) to trigger CI
- `bun run validate` green (lint + type-check + prettier)

## Next

- Approve CI workflows (currently in `action_required` state)
- Address the **blocking issue**: remove `[skip ci]` empty commit logic from `pushBranch` (action triggers on `issue_comment`/ `pull_request_review`, never on push)
- Consider addressing non-blocking issues before merge:
  - Pin peer deps (`@earendil-works/pi-coding-agent: "^0.79.6"`, `typebox: "^1.2.14"`)
  - Sync bridge version to match monorepo (`2.20.3`)
- After merge, follow up on remaining non-blocking items (pagination optimization, double error notification, Fallow complexity findings)

**Comment by shaftoe** [comment]
/pi 🤖 Handoff

## Done

- Updated packages/pi-action-bridge/README.md to remove previous design details
- Updated AGENTS.md to simplify the bridge section and document custom prompts approach

## Next

- Address blocking issue: remove `[skip ci]` empty commit logic from `pushBranch`
- Consider addressing non-blocking issues before merge (pin peer deps, sync version)
- Approve CI workflows to run full validation + test suite

**Comment by shaftoe** [comment]
/pi apply recommendations from the latest review, push changes back into the pr

**Comment by shaftoe** [comment]
/pi 🤖 Handoff

## Done

Since the last reviewed commit (`ff70e28`), the bridge grew from 1 tool to 3:

- **Added two new Octokit-backed tools**: `post_pr_comment` and `read_pr_thread` (registered in `index.ts`, public symbols re-exported). The bridge now exposes the full `/handoff` + `/pickup` loop natively instead of leaving it to prompt prose.
- **Expanded `octokit.ts` (+197 lines)** with the I/O layer for both: `postIssueComment`, `readIssueThread`, `normalizeComment` projection, plus factory variants `createOctokitPostComment` / `createOctokitReadThread` / `createOctokitFromEnv`.
- **Expanded `types.ts`**: added `NormalizedComment`, `NormalizedThread`, `PostPrCommentDetails`, `ReadPrThreadDetails`.
- **Rewrote `prompts/handoff.md` + `prompts/pickup.md`** to route through the new `post_pr_comment` / `read_pr_thread` tools. This drops the unconditional `GIT_CONFIG_GLOBAL=/dev/null` git-env prefix that previously leaked into `/pickup`'s `git pull` — **addresses review #12 finding #1** (the git-config footgun for local TUI).
- **Removed dead `toRepoInfo` from `remote.ts`** — **addresses review #12 finding #2**.
- **Added tests**: `tests/octokit.spec.ts` (346 lines), `tests/post-pr-comment.spec.ts` (89), `tests/read-pr-thread.spec.ts` (142). Bridge test count: **67 → 101**, all passing.
- `bun run validate` green (lint + type-check + prettier).

## Next

- **History cleanup**: the two new commits (`74d1bd0`, `17fa132`) are both `wip` and already pushed — squash/reword before merge.
- **Review needed on the two new tools**: `post_pr_comment` and `read_pr_thread` (and the expanded `octokit.ts`) have not been reviewed yet — only `detect_pull_request` has.
- **Carry-over non-blockers still open**: version drift (`2.20.3` vs monorepo `2.20.4`, auto-healed at release by `sync-versions.ts`); Fallow duplication of `resolveGitHubToken` across `pi-action-bridge` / `pi-cli`.
- Confirm the now-~515-line `octokit.ts` doesn't worsen the existing Fallow code-duplication hotspot flagged on this PR.

**Comment by shaftoe** [comment]
/pi 🤖 Handoff

## Done

Drove a Fallow pass against `packages/pi-action-bridge/` to kill the duplication + complexity smells flagged on this PR (committed as `9090e94`, pushed — `0 ahead/0 behind origin/newBridge`):

- **Killed bridge-internal clone `dup:e919f3f9` (64 lines)** — `post_pr_comment` + `read_pr_thread` each re-declared `DEFAULT_SERVER_URL`, `resolveServerUrl`, and identical `owner`/`repo`/`number`/`server_url` param fields. Extracted into new `packages/pi-action-bridge/src/tools/common.ts` (`ownerField`, `repoField`, `serverUrlField`, `prNumberField(desc)`, `resolveServerUrl`, `DEFAULT_SERVER_URL`); both tool files now compose schemas from it. Also removed the cross-file `import { DEFAULT_SERVER_URL } from './post-pr-comment'` smell.
- **Killed cross-package clone `dup:abb7a441` (55 lines)** — `resolveGitHubToken` was byte-for-byte duplicated between `pi-action-bridge/src/octokit.ts` and `pi-cli/src/auth.ts`. Promoted the canonical copy to new `packages/pi-platform-github/src/auth.ts` (both packages already depend on it), re-exported from the barrel; `pi-cli/src/auth.ts` is now a thin re-export, the bridge imports it. Dropped the duplicate test blocks from both packages; added canonical coverage in `packages/pi-platform-github/tests/auth.spec.ts`.
- **Resolved the `readIssueThread` complexity hotspot** (was 65 lines, cyclomatic 20, cognitive 20): added a `requestOpts(signal)` helper for the `...(signal ? {request:{signal}}:{})` ternary repeated 5×; split the function into `clampMax` / `issueState` / `fetchPrMeta` / `fetchComments`; dropped a no-op `if (state==='closed' && pr.merged_at)` branch.
- **Fallow delta:** bridge `dupes` 2 clone groups → 0 (duplication 9.8% → 0.0%); repo-wide `dupes` no longer reports the `resolveGitHubToken` clone (only a pre-existing, out-of-scope `pi-action`/`pi-cli` adapter clone remains); bridge `health` "Large functions" + "High complexity functions" sections gone; score 75 → 80, avg cyclomatic 3.5 → 3.0, p90 8 → 7, maintainability 92.3 → 93.0; `dead-code` clean.
- `bun run validate` green; **1501 pass / 0 fail** (pi-platform-github +6 new auth tests; pi-action-bridge −6 duplicate tests; pi-cli −5 — all canonical behavior still covered).

## Next

- **History cleanup:** `9090e94` is the third consecutive `wip` commit on this branch (`74d1bd0`, `17fa132`, `9090e94`) — squash/reword before merge.
- **Review the cross-package promotion:** moving `resolveGitHubToken` into `pi-platform-github` touches a shared package (3 files there + the barrel export); worth a reviewer's eye since it affects `pi-cli`'s import surface too.
- **Carry-over non-blockers still open:** version drift (`2.20.3` bridge vs monorepo `2.20.4`, auto-healed at release by `sync-versions.ts`); the unrelated pre-existing clone in `pi-action/src/adapters/pi-agent-adapter.ts` ↔ `pi-cli/src/adapters/pi-agent.ts` (Fallow `dup:93c949a0`, 17 lines — not introduced here, could fold into a separate cleanup).

File: packages/pi-action-bridge/src/octokit.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+ *
+ * Builds an Octokit instance (matching the structural shape used by pi-cli and
+ * `pi-platform-github`) and implements the three I/O operations the bridge
+ * tools need: PR lookup, comment posting, and thread reading. The forge-
+ * specific REST base URL is derived via the reused `apiBaseUrlFromServerUrl`,
+ * giving us GitHub / GHES / Codeberg / Forgejo / Gitea parity for free.
+ *
+ * Token resolution is the shared `resolveGitHubToken` from
+ * `pi-platform-github` (`GITHUB_TOKEN` → `GH_TOKEN`). There is no `gh` CLI
+ * fallback for these tools (the `handoff`/`pickup` prompts already cover `gh`).
```
[the remark this answers is not in the document]
Comment (shaftoe) [reply packages/pi-action-bridge/src/octokit.ts]: /pi fix the document comment

[the remark this answers is not in the document]
Comment (shaftoe) [reply packages/pi-action-bridge/src/octokit.ts]: /pi fix the document comment

**Comment by shaftoe** [comment]
/pi review one more time this pull request, suggest improvements

**Comment by shaftoe** [comment]
/pi apply recommendations from "Re-review (head 72abc87)", push back into the pr

**Comment by shaftoe** [comment]
/pi update packages/pi-action-bridge/package.json to latests table versions, push changes back into the PR

**Comment by shaftoe** [comment]
/pi update both readme (clean it up) and package.json (use latest stable deps as develop branch does), rebase on latest develop branch too, push all back into the pr

**Comment by shaftoe** [comment]
/pi fix the "Three minor non-blocking carry-overs remain (plain git pull → --ff-only, sequential API calls → Promise.all, comment cap 50 → 100), plus the pre-merge history squash.", push changes back into the PR
